src/Nest/CommonAbstractions/SerializationBehavior/SerializerFactory.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Elasticsearch.Net;
+using Newtonsoft.Json;
+
+namespace Nest
+{
+	public interface ISerializerFactory
+	{
+		IElasticsearchSerializer Create(IConnectionSettingsValues settings);
+
+		IElasticsearchSerializer CreateStateful(IConnectionSettingsValues settings, JsonConverter converter);
```

Comment:
I'm wondering if we really need an additional method for this.  Can we just add `JsonConverter` as a nullable parameter on `Create`?  Stateful converters are really an internal thing, and I think it's best to "hide" it from the user.